### ISSUE TITLE
File system interaction

### ISSUE THREAD

**Issue by coatless** [rstudio/rscloud#15]
Any chance there could be a file system component to the API? 

In particular, the interactions I'm thinking of are:

1. automatically downloading a `.zip` of active student files 
2. adding and overwriting files

**Comment by tareefk** [comment]
Just to clarify, when you say: Adding or overwriting files, are you looking for a programmatic way to replace the contents of a student's existing content?

**Comment by coatless** [comment]
Yes, I'm looking for a programmatic way to easily fix a "bad" data set if it was released or directly send grades back to the student project.

**Comment by tareefk** [comment]
The dataset problem we will solve a different way, we may not have talked about it but our goal is to have a directory that gets mounted as read only which you could be changing all the time underneath the hood.

**Comment by wlandau** [comment]
Maybe related: is it possible to modify the *upstream* project? I would like to continuously deploy https://rstudio.cloud/project/627076 from https://github.com/wlandau/learndrake/tree/master/inst/notebooks.

**Comment by kevinykuo** [comment]
@wlandau Do you mind opening an issue to make sure we track it? That's something I wondered about, too.

**Comment by wlandau** [comment]
Sure: #35
